Material.Blazor/Foundation/DialogAwareBatchingJSRuntime.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Threading.Tasks;
+
+namespace Material.Blazor.Internal
+{
+    internal class DialogAwareBatchingJSRuntime : IBatchingJSRuntime
+    {
+        private readonly IBatchingJSRuntime underlyingJSRuntime;
+        private readonly MBDialog dialog;
+        public DialogAwareBatchingJSRuntime(IBatchingJSRuntime underlyingJSRuntime, MBDialog dialog)
+        {
+            this.underlyingJSRuntime = underlyingJSRuntime;
+            this.dialog = dialog;
+        }
+        public async Task<T> InvokeAsync<T>(string identifier, params object[] args)
+        {
+            await dialog.Opened;
```

Comment:
For a JS call on a component that is within a dialog, this code now handles the JS interop, so we will always wait until the dialog is actually open.